src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -176,10 +215,29 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 
             bool remapPorts = true;
 
-            // If type is still null at this point return a dummy node
             if (type == null)
             {
-                node = CreateDummyNode(obj, typeName, assemblyName, functionName, inPorts, outPorts);
+                if (typeName == "CoreNodeModels.Formula")
```

Comment:
I'm fine with this approach for now, it might be nice to eventually refactor deserialization to have a JSON -> JSON migration function - we don't need to convert the existing migration system to support JSON graphs, but better separation between normal deserialization and "migration" would probably be good for readability.